src/platform/AT32/timer_at32bsp.c

```diff
@@ -777,4 +777,50 @@ uint16_t timerGetPrescalerByDesiredHertz(tmr_type *tim, uint32_t hz)
     }
     return (uint16_t)((timerClock(tim) + hz / 2 ) / hz) - 1;
 }
+
+void timerReset(tmr_type *timer)
+{
+    ATOMIC_BLOCK(NVIC_PRIO_TIMER) {
```

Comment:
A comment as to why you need this to be atomic whereas you don’t with other processors would be helpful.

Reply:
It may actually not be necessary - it was in the original driver - so I have left it. Be worth testing whether it makes a difference.

Reply:
@blckmn: Testing won't help here, race condition probability is very low, it is unlikely it will happen during test. But it can happen quite often over all BF installations.
But in this case it only does `tmr_x->ctrl1_bit.tmren = new_state;`, it is safe with interrupts enabled.

Only possible problem is that ATOMIC_BLOCK adds memory barrier, which may affect code (invalid one - with undefined behavior) calling this function.